### ISSUE TITLE
Running dune promote selectively

### ISSUE THREAD

**Issue by andreypopp** [ocaml/dune#1348]
TL;DR: would be nice to have `dune runtest --promote`

I don't usually remember `dune` invocations and sometimes trying to figure them out.

So today I was working with `dune runtest` and dune's promote feature.

1. I ran `esy dune runtest ./esyi` and got failures which can be promoted
2. I instinctively ran `esy dune runtest --promote ./esyi` which didn't work (no such option `--promote`)
3. Then I looked up docs and found out it's `promote` command, so I ran `esy dune promote ./esyi` which said `Warning: Nothing to promote for esyi`
4. Finally I dropped `./esy` and ran `esy dune promote` which worked.

Thanks!

**Comment by emillon** [comment]
Hi,

There's a `--auto-promote` option that is supported by most commands. In that case, `dune runtest DIR --auto-promote` should do the trick. It's mostly equivalent to running `dune runtest DIR; dune promote`.

The confusing part is that `dune promote` takes files as arguments, not directories or aliases. So you'd have to pass the actual files that you want to be promoted.

Does `--auto-promote` solve your problem?
